py_ecc/optimized_bls12_381/optimized_swu.py

```diff
@@ -15,11 +16,51 @@
     ISO_3_B,
     ISO_3_MAP_COEFFICIENTS,
     ISO_3_Z,
+    ISO_11_A,
+    ISO_11_B,
+    ISO_11_MAP_COEFFICIENTS,
+    ISO_11_Z,
+    P_MINUS_3_DIV_4,
     P_MINUS_9_DIV_16,
     POSITIVE_EIGHTH_ROOTS_OF_UNITY,
+    SQRT_MINUS_11_CUBED,
 )
 
 
+# Optimized SWU Map - FQ to G1'
+# Found in Section 4 of https://eprint.iacr.org/2019/403
+def optimized_swu_G1(t: FQ) -> Tuple[FQ, FQ, FQ]:
+    t2 = t**2
+    iso_11_z_t2 = ISO_11_Z * t2
+    temp = iso_11_z_t2 + iso_11_z_t2**2
+    denominator = -(ISO_11_A * temp)  # -a(Z * t^2 + Z^2 * t^4)
```

Comment:
Inline comments here and on line 38 are a little confusing, it doesn't seem to line up with the actual code, are these still relevant?

Reply:
I think it is. Lines 33 and 34 create `Z * t^2`. Line 35 squares the result (which is `Z^2*t^4`) and adds it to the original result. All that is then multiplied by negative `ISO_11_A`, which is the `-a`. Line 38 adds 1 to the original result, then multiplies by `b`.

Reply:
Ok

Reply:
Ah thanks, hadn't noticed the other vars above line 36 were components of the whole equation. I think the inline comment threw me off but it's fine as is.